File: deezer_history_bot/handlers/status.py
from aiogram.types import Message
from httpx import AsyncClient

from .. import get
from ..access import get_history
from ..errors import errors
from ..image import create_image

httpx = AsyncClient()


@errors
async def handler(message: Message):
    if message.chat.type == 'private':
        await message.reply('You should send this in a group!')
    else:
        track: dict = (await get_history(message.from_user.id))[get.indent(message)]
        await message.reply_photo(
            photo=await create_image(track, message.from_user),
            reply_markup=get.lt_reply_markup(track),
        )

File: src/handlers/status.ts
import { Composer } from "grammy";
import { getHistory } from "../history";
import { getAccess } from "../access";
import { getImage } from "../image";
import { getReplyMarkup } from "../helpers";

export default new Composer().command("status", async (ctx) => {
  if (ctx.chat.type == "private" || !ctx.from || !ctx.message) {
    await ctx.reply("You should send this in a group!");
    return;
  }
  const access = await getAccess(ctx.from.id);
  if (access === "") {
    await ctx.reply(
      "You need to connect your Deezer account first. PM me and use the /connect command.",
      {
        reply_markup: {
          inline_keyboard: [
            [
              {
                text: "PM me",
                url: `https://t.me/${ctx.me.username}`,
              },
            ],
          ],
        },
      }
    );
    return;
  }
  var indent = parseInt(ctx.message.text.split(/\s/g)[1]);
  const history = await getHistory(access);
  if (!(indent - 1 in history)) indent = 0;
  else indent -= 1;
  const track = history[indent];
  await ctx.replyWithPhoto(
    {
      file: await getImage(
        track.album.cover_big,
        ctx.from.first_name,
        track.title,
        track.artist.name,
        track.album.title
      ),
    },
    { reply_markup: getReplyMarkup(track) }
  );
});
